MapzenSDK/MZMapViewController.swift

```diff
@@ -487,8 +487,10 @@ open class MZMapViewController: UIViewController, LocationManagerDelegate {
    - parameter marker: The marker to add to map.
    */
   open func addMarker(_ marker: GenericMarker) {
-    currentMarkers[marker.tgMarker] = marker
-    marker.tgMarker.map = tgViewController
+    if marker.tgMarker == nil {
+      marker.tgMarker = tgViewController.markerAdd()
+    }
+    currentMarkers[marker.tgMarker!] = marker
```

Comment:
It's a slightly different use case. With this change we're effectively decoupling SDK markers from Tangram-es markers, meaning we no longer require a Tangram-es marker be created at initialization time of the SDK marker class.  We're also allowing users to set their own underlying TGMarkers for their SDK markers if they want (it honestly makes it easier on us to implement as well).

The side effect of all that is we need to check we don't override previous markers when adding the SDK markers to the map.